docs/deep-dive/scaffolding/your-first-scaffold-plugin.md

```diff
@@ -0,0 +1,138 @@
+
+Let's see how **webiny.root.js** changes after adding our plugin:
+
+- Our original file
+```js
+module.exports = {
+    projectName: "webiny-js",
+    cli: {
+        plugins: [
+            "@webiny/cli-plugin-deploy-components",
+            "@webiny/cli-plugin-scaffold",
+            "@webiny/cli-plugin-scaffold-graphql-service",
+            "@webiny/cli-plugin-scaffold-lambda"
+        ]
+    }
+};
+```
+- **Local / Global package** (which exports one or multiple plugins)
+```js
+module.exports = {
+    projectName: "webiny-js",
+    cli: {
+        plugins: [
+            "@webiny/cli-plugin-deploy-components",
+            "@webiny/cli-plugin-scaffold",
+            "@webiny/cli-plugin-scaffold-graphql-service",
+            "@webiny/cli-plugin-scaffold-lambda",
+            "./scaffoldFolder",
+        ]
+    }
+};
+```
+- **Inline plugin**
+```js
+module.exports = {
+    projectName: "webiny-js",
+    cli: {
+        plugins: [
+            "@webiny/cli-plugin-deploy-components",
+            "@webiny/cli-plugin-scaffold",
+            "@webiny/cli-plugin-scaffold-graphql-service",
+            "@webiny/cli-plugin-scaffold-lambda",
+            {
+                name: "cli-plugin-scaffold-template-scaffold-folder",
+                type: "cli-plugin-scaffold-template",
+                scaffold: {
+                    name: "Scaffold folder",
+                    questions: () => [
+                        /* ... */
+                    ],
+                    generate: () => {
+                        /* ... */
+                    }
+                }
+            }
+        ]
+    }
+};
+```
+
+# Scaffold plugin structure
+As you may see in the inline example above, the plugin's **type** must always be **cli-plugin-scaffold-template**.
+But more signifitcantly, you'll provide a **scaffold** object that contains three fields:
+- **name** - This is the name of your scaffolding plugin as it will appear in the CLI.
+- **questions({ context })** -  A function that returns an array of [inquirer](https://www.npmjs.com/package/inquirer)-like questions.
+- **generate({ context, input, oraSpinner })** -  Finally, your actual scaffolding code. This function gets executed after correctly answering all **questions** above.
+    - **input** - An object containing the inputted values received from your inquirer questions
+    - **oraSpinner** - Remember the pretty spinner you see when interacting with our CLI? This is the [Ora Spinner](https://www.npmjs.com/package/ora) instance for you to use in case you need to update it.
+
+The **context** object referenced above contains various data that might be helpful: certain paths, the config file's data, registered plugins, ... . Feel free to output it & take a deeper look at it.
+
+
+# Example
+Let's create our first scaffolding plugin. I've navigated to my project's root and created a new folder: `./scaffoldFolder`. Here I'll quickly initialise a new package using `npm init -y`.
```

Comment:
IMHO
`First of all I'll navigate` instead of `I've navigated`

Reply:
done